docs/structured-streaming-programming-guide.md

```diff
@@ -517,6 +517,8 @@ There are a few built-in sources.
 
   - **Rate source (for testing)** - Generates data at the specified number of rows per second, each output row contains a `timestamp` and `value`. Where `timestamp` is a `Timestamp` type containing the time of message dispatch, and `value` is of `Long` type containing the message count, starting from 0 as the first row. This source is intended for testing and benchmarking.
```

Comment:
A little concern about the confusion between "rate" and "rate per micro-batch". For original "rate" data source, "Generates data at the specified number of rows per second," it sounds like the rate of data generation is fixed, so mostly it makes the assumption that the number of rows per micro-batch is the same or close. Could we put some wordings into "rate" or "rate per micro-batch" that clarifies the difference for end-users?

Reply:
I'd love to address it, but honestly I have no idea other than below representation:

* rate: Generates data at the specified number of rows per second
* rate per micro-batch: Generates data at the specified number of rows per micro-batch

Specified number of rows "per XXX" says the specified number of rows will be presented per XXX, so the main point they should check is the unit (per XXX). "per second" vs "per micro-batch" doesn't seem to make confusion IMHO.

Reply:
I'm open to alternative names for the data source as well. Picking up "rate" seems to be culprit, but I can't think of alternative names.

Reply:
The name of data source looks okay to me.

Maybe you can copy some wordings from the java doc here too, e.g., "Unlike `rate` data source, this data source provides a consistent set of input rows ...".